inspirehep/submissions/marshmallow/author.py

```diff
@@ -0,0 +1,208 @@
+    family_name = fields.Raw()
+    given_name = fields.Raw()
+    native_name = fields.Raw()
+    public_emails = fields.Raw()
+    orcid = fields.Raw()
+
+    status = fields.Raw()
+    arxiv_categories = fields.Raw()
+    websites = fields.Raw()
+    twitter = fields.Raw()
+    blog = fields.Raw()
+    linkedin = fields.Raw()
+
+    positions = fields.Raw()
+    project_membership = fields.Raw()
+    advisors = fields.Raw()
+
+    comments = fields.Raw()
+
+    acquisition_source = fields.Raw()
+    bai = fields.Raw()
+
+    @pre_dump
+    def before_dump(self, data):
+        family_name, given_name = self.get_name_split(data)
+        return {
+            "advisors": get_value(data, "advisors", default=missing),
+            "acquisition_source": get_value(
+                data, "acquisition_source", default=missing
+            ),
+            "arxiv_categories": get_value(data, "arxiv_categories", default=missing),
+            "blog": self.get_first_or_missing(
+                self.get_value_by_description_key(data.get("urls", []), "blog")
+            ),
+            "display_name": get_value(data, "name.preferred_name", default=missing),
+            "family_name": self.get_value_or_missing(family_name),
+            "given_name": self.get_value_or_missing(given_name),
+            "linkedin": self.get_first_or_missing(
+                get_values_for_schema(data.get("ids", []), "LINKEDIN")
+            ),
+            "native_name": get_value(data, "name.native_names[0]", default=missing),
+            "orcid": self.get_first_or_missing(
+                get_values_for_schema(data.get("ids", []), "ORCID")
+            ),
+            "positions": get_value(data, "positions", default=missing),
+            "project_membership": get_value(
+                data, "project_membership", default=missing
+            ),
+            "public_emails": get_value(data, "email_addresses.value", default=missing),
+            "status": get_value(data, "status", default=missing),
+            "twitter": self.get_first_or_missing(
+                get_values_for_schema(data.get("ids", []), "TWITTER")
+            ),
+            "websites": get_value(data, "urls.value", default=missing),
+        }
+
+    @staticmethod
+    def get_name_split(data):
+        name = get_value(data, "name.value")
+        if not name:
+            return missing, missing
+
+        name = name.split(",")
+        if len(name) > 1:
+            return name[0].strip(), name[1].strip()
+        return missing, name[0].strip()
+
+    @staticmethod
+    def get_value_or_missing(value):
+        if value:
+            return value
+        return missing
+
+    @staticmethod
+    def get_first_or_missing(value):
+        if value:
+            return value.pop()
+        return missing
+
+    @staticmethod
+    def get_full_name(family_name, given_name):
+        if given_name and family_name:
+            return u"{}, {}".format(family_name, given_name)
```

Comment:
we can use ``f-Strings`` from python 3 <https://realpython.com/python-f-strings/>